### ISSUE TITLE
[Bug]: Querying for features based on transformations with entity name throws errors

### ISSUE THREAD

**Issue by samuell** [featureform/featureform#437]
### Summary

I get errors when I try to query features that are registered upon a `df_transformation` . It seems the entity name does not work there. If I use the entity *column name* instead, I don't get errors anymore, and actually get some output, although it is not correct. See below, and also [this folder in a separate repo](https://github.com/samuell/bugs/tree/main/02-ff-entity-match) for all the code needed to reproduce this.

### Expected Behavior

Should get this output:

```
-------Foo-------
0.002
-------Bar-------
0.004
-------FooPlusBar-------
0.006
```

### Actual Behavior

I get this output:

```
-------Foo-------
0.002
-------Bar-------
0.004
-------FooPlusBar-------
Traceback (most recent call last):
  File "/home/sal/proj/sav/2022/bug-reproductions/02-ff-entity-match/client.py", line 15, in <module>
    fooplusbar = client.features([("fooplusbar", "default")], {"person": "samuel"})
  File "/home/sal/.cache/pypoetry/virtualenvs/20220903-ff-entity-match-9koWNEX7-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 104, in features
    return self.impl.features(features, entities)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220903-ff-entity-match-9koWNEX7-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 304, in features
    all_features_list = self.add_feature_dfs_to_list(feature_variant_list, entity_id)
  File "/home/sal/.cache/pypoetry/virtualenvs/20220903-ff-entity-match-9koWNEX7-py3.9/lib/python3.9/site-packages/featureform/serving.py", line 319, in add_feature_dfs_to_list
    raise ValueError(
ValueError: Could not set entity column. No column name person exists in compute_fooplusbar-default
```

### Steps To Reproduce

The code to reproduce this is available [in this repo](https://github.com/samuell/bugs/tree/main/02-ff-entity-match), but adding the reproduction info here as well:

Save this data to file named `data.csv`:

```
time,foo,bar,person_id
2022-08-25 00:00:01,0.000,0.001,samuel
2022-08-25 00:00:02,0.001,0.002,samuel
2022-08-25 00:00:03,0.002,0.004,samuel
```

Put this in `defs.py`:

```python
import featureform as ff

ff.register_user("myself").make_default_owner()

local = ff.register_local()

dummydata = local.register_file(
    name="dummydata",
    variant="default",
    description="",
    path="data.csv",
)

person = ff.register_entity("person")

dummydata.register_resources(
    entity="person",
    entity_column="person_id",
    inference_store=local,
    features=[
        {
            "name": "foo",
            "variant": "default",
            "column": "foo",
            "type": "float32",
        },
        {
            "name": "bar",
            "variant": "default",
            "column": "bar",
            "type": "float32",
        },
    ],
    timestamp_column="time",
)

@local.df_transformation(variant="default", inputs=[("dummydata", "default")])
def compute_fooplusbar(df):
    df["fooplusbar"] = df["foo"] + df["bar"]
    return df

compute_fooplusbar.register_resources(
    entity=person,
    entity_column="person_id",
    inference_store=local,
    features=[
        {
            "name": "fooplusbar",
            "variant": "default",
            "column": "fooplusbar",
            "type": "float32",
        },
    ],
    labels=[
        {
            "name": "fooplusbar",
            "variant": "default",
            "column": "fooplusbar",
            "type": "float32",
        },
    ],
    timestamp_column="time",
)
```

Put this in `client.py`:

```python
import featureform as ff

client = ff.ServingClient(local=True)

print("-"*7 + "Foo" + "-"*7)
foo = client.features([("foo", "default")], {"person": "samuel"})
print(f"{foo[0]:.3f}")

print("-"*7 + "Bar" + "-"*7)
bar = client.features([("bar", "default")], {"person": "samuel"})
print(f"{bar[0]:.3f}")

print("-"*7 + "FooPlusBar" + "-"*7)
fooplusbar = client.features([("fooplusbar", "default")], {"person": "samuel"})
print(f"{fooplusbar[0]:.3f}")
```

Run:
```bash
featureform apply --local defs.py
python client.py
```

### What mode are you running Featureform in?

Local

### What version of Python are you running?

3.9

### Featureform Python Package Version

1.1.12

### Other info

If using the entity column name (`person_id`) instead of the entity name
(`person`) for the feature based on a transformation ... that is, putting this code into `client_works_but_gives_wrong_value.py` 

```python
import featureform as ff

client = ff.ServingClient(local=True)

print("-"*7 + "Foo" + "-"*7)
foo = client.features([("foo", "default")], {"person": "samuel"})
print(f"{foo[0]:.3f}")

print("-"*7 + "Bar" + "-"*7)
bar = client.features([("bar", "default")], {"person": "samuel"})
print(f"{bar[0]:.3f}")

print("-"*7 + "FooPlusBar" + "-"*7)
# *** NOTE below that we use "person_id" (the entity COLUMN NAME) instead of "person" (the entity NAME): ***
fooplusbar = client.features([("fooplusbar", "default")], {"person_id": "samuel"})
print(f"{fooplusbar[0]:.3f}")
```

... and executes it with python ... then, we get output, but not
correct values:

**Expected output:**

```bash
-------Foo-------
0.002
-------Bar-------
0.004
-------FooPlusBar-------
0.006
```

**Actual output**

```bash
-------Foo-------
0.002
-------Bar-------
0.004
-------FooPlusBar-------
0.001
```

**Comment by sdreyer** [comment]
Hi @samuell , I have a fix in with prerelease version 1.1.13rc0. I'll add additional testing and push an official release when we're back on Tuesday. I believe this will work for now if this is blocking.

**Comment by samuell** [comment]
@sdreyer Awesome, thanks!
